### PR TITLE
Optional snake case table names

### PR DESCRIPTION
**Pull request by waffletower-reify**
Here is a PR for our local kafka-connect-iceberg-sink repo implementing snake_case coercion of table identifiers.

Below follows getindata PR template:

#### Description

`ADDED optional snake_case coercion for Iceberg table identifiers`

Allows for interoperability with case-insensitive databases such as Trino

##### PR Checklist
- [x] Tests added
- [x] [Changelog](CHANGELOG.md) updated

### CODE REVIEW

File: src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeConsumer.java [hunk 1]
```diff
@@ -46,7 +46,11 @@ public void accept(Collection<SinkRecord> records) {
                 .collect(Collectors.groupingBy(IcebergChangeEvent::destinationTable));
 
         for (Map.Entry<String, List<IcebergChangeEvent>> event : result.entrySet()) {
-            TableIdentifier tableIdentifier = TableIdentifier.of(Namespace.of(configuration.getTableNamespace()), configuration.getTablePrefix() + event.getKey());
+            String eventKey = event.getKey();
+            if (configuration.isTableSnakeCase() == true) {
```
Comment (dacreify) [inline src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeConsumer.java]:
```suggestion
            if (configuration.isTableSnakeCase()) {
```

File: src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergUtil.java [hunk 2]
```diff
@@ -89,4 +89,44 @@ public static GenericAppenderFactory getTableAppender(Table icebergTable) {
...
+      
+      StringBuilder sb = new StringBuilder();
+      boolean lastUpper = true;
+      boolean lastSeparator = false;
+
+      for (Character c : inputString.toCharArray()) {
+          
+          if (Character.isUpperCase(c)) {
+
+              if (lastUpper == false) {
```
Comment (dacreify) [inline src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergUtil.java]:
```suggestion
              if (! lastUpper) {
```

File: src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergUtil.java [hunk 3]
```diff
@@ -89,4 +89,44 @@ public static GenericAppenderFactory getTableAppender(Table icebergTable) {
...
+
+                  if (lastSeparator == false) {
```
Comment (dacreify) [inline src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergUtil.java]:
```suggestion
                  if (! lastSeparator) {
```

**Review by dacreify — APPROVED** [review]
Looks good!

File: src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergUtil.java (hunk 3 above)
Comment (waffletower-reify) [reply src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergUtil.java]: Thanks, can use ancient C habits here instead.

**Pull request merged**